pennylane/gradients/spsa_gradient.py

```diff
@@ -300,6 +303,29 @@ def spsa_grad(
         # Skip the unshifted tape
         shifts = shifts[1:]
 
+    if not isinstance(sampler_rng, (int, np.random.Generator, type(None))):
+        raise ValueError("Argument sampler_rng has invalid type.")
```

Comment:
```suggestion
        raise ValueError(f"The argument sampler_rng is expected to be a NumPy PRNG, an integer or None, but is {sampler_rng}.")
```

Reply:
Should we do `"... but is {type(sampler_rng)}."` instead?

Reply:
I was thinking so too. I'm not sure what's the more helpful situation here. One reason I saw against printing the `type` was that an object of type `A` can be an instance of a parent class `B`, detaching the printed info a bit from the actual test that leads to the error :thinking: 
However, it probably does not matter too much and if you find printing the `type` more clear, go for it :)

Reply:
Oh, that's a good point. I'll add it as you suggested.